### PR TITLE
Fix crash/Error in techmenu with 0 intel entries.

### PR DESCRIPTION
**Pull request by rtoijala**
`tech_render_anim` did not check whether there were any intel/weapon
entries before trying to read from`Current_list`, which means that it
read at `Current_list[-1]`, which is a Bad Thing (TM).

Do not try to render an anim if there is no entry.

### CODE REVIEW

**Comment by asarium** [comment]
When does this happen? It's always a good thing to validate parameters but if the code calling this function is using the wrong parameters then it might be better to fix that code and use an Assertion here.

If that is not the case then the code is good to go.

**Comment by rtoijala** [comment]
It happens when overriding `species.tbl` with one that does not contain any intel entries, i.e. an empty file. Also, when `species.tbl` is completely missing (which IMHO should be treated like an empty file). Also, when `species.tbl` contains only entries that are not displayed by default (and none have been unlocked).

tl;dr when there are no entries to show.

`techroom_do_frame` calls `techroom_anim_render` unconditionally when the intel tab is selected. If there are no intel entries, `Current_list_size` is 0 and `Cur_entry` is -1.

**Comment by asarium** [comment]
Ok, that sounds like it's not a bug or data error.
